src/redux/auth/index.js

```diff
@@ -0,0 +1,44 @@
+import API from '../api';
+
+export const loadCurrentUser = (state) => state.auth.currentUser;
+
+const LOAD_CURRENT_USER = 'auth/current_user';
+const RESET = 'auth/reset';
+
+export const setCurrentUser = (payload) => ({
+  type: LOAD_CURRENT_USER,
+  payload,
+});
+
+export const login = (payload) => async (dispatch) => {
+  const response = await API.login(payload);
+  dispatch(setCurrentUser(response));
+};
```

Comment:
The API.login function has a callback function that returns a response. Therefore the dispatch should be inside that call back to be able to return the appropriate response. Example
```
export const login = (payload) => async (dispatch) => {
  await API.login(payload, (response) => {
    dispatch(setCurrentUser(response.data));
  });
};
```

It would be best if you did the same for the `sign_up` function.